### ISSUE TITLE
Test Suite and CI Python 3 Support

### ISSUE THREAD

**Issue by pmart123** [psycopg/psycopg2#460]
I have started working on a [fork](https://github.com/pmart123/psycopg2/tree/tests-python3) that ports the test suite for Python 2.7/3.5 compatibility. Before I carry on, I want to know if the core contributors had anything they would want or find beneficial? Ideally, I would setup the CI and test suite to support and run against different Postgres and Python versions. Personally, I think the test suite should drop support for unsupported versions of Python and Postgres, and to have the test suite reflect this.

**Comment by dvarrazzo** [comment]
Hi, we have a [test grid](http://initd.org/buildbot/tgrid) for all the supported versions, running the test suite from py 2.5 to 3.5 and postgres from 7.4 to 9.5, both in sync and async mode. 

Support for Python 2.5 is planned to be dropped in psycopg 2.7. I think I would also drop support for unmaintained libpq versions to build that version, because we need too many features to ifdef them out... and because unsupported. Not really planning to kill 2.6 yet for the moment.

What is exactly that you want to achieve?

**Comment by pmart123** [comment]
Ok. Thanks for the clarification. I noticed you recently dropped support for unmaintained libq versions/python 2.5, which was the step I ultimately hoped the maintainers were moving towards. I also noticed some formatting, old vs. new conventions for Python in the test suite, and was willing to help modernize/cleanup some of these.

<img width="736" alt="screen shot 2016-08-31 at 8 57 06 pm" src="https://cloud.githubusercontent.com/assets/8962926/18151494/8e5b57f4-6fbd-11e6-959d-6bc2930eb2e2.png">
